src/backend/joanie/core/management/commands/create_certificates.py

```diff
@@ -0,0 +1,223 @@
+        ]
+
+        certificate_counter = 0
+        for order in orders:
+            result = self._generate_certificate_for_order(order)
+            certificate_counter += result
+
+        if certificate_counter > 0:
+            logger.info(
+                '%s certificate(s) for "%s" have been generated.',
+                certificate_counter,
+                product.title,
+            )
+
+        return certificate_counter
+
+    @staticmethod
+    def _generate_certificate_for_order(order):
+        """
+        Check if order is eligible for certification then generate certificate is it is.
+
+        Eligibility means that order contains
+        one passed enrollment per graded courses.
+
+        Return:
+            0 : if the order is not eligible to certification
+            1: if a certificate has been generated for the current order
+        """
+        graded_courses = order.target_courses.filter(
+            order_relations__is_graded=True
+        ).order_by("order_relations__position")
+        graded_courses_count = len(graded_courses)
+
+        if graded_courses_count == 0:
+            return 0
+
+        course_runs = models.CourseRun.objects.filter(
+            course__in=graded_courses,
+            is_gradable=True,
+            start__lte=timezone.now(),
+        )
+
+        enrollments = order.enrollments.filter(
+            course_run__in=course_runs, is_active=True
+        ).select_related("user", "course_run")
+
+        # Cross graded courses and enrollments to check there is an active enrollment
+        # for each graded courses, if not it is useless to go further
+        course_enrollments = []
+        for course in graded_courses:
+            for enrollment in enrollments:
+                # Check if the enrollment relies on course by crossing
+                # all course runs implied
+                intersections = len(
+                    {enrollment.course_run} & set(course.course_runs.all())
+                )
+                if intersections == 1:
+                    course_enrollments.append(enrollment)
+                    break
+
+        # If we do not have one enrollment per graded course, there is no need to
+        # continue, we are sure that order is not eligible for certification.
+        if len(course_enrollments) != graded_courses_count:
+            return 0
+
+        # Otherwise, we now need to know if each enrollment has been passed
+        passed_enrollment_count = 0
+        for enrollment in course_enrollments:
+            if enrollment.is_passed is False:
+                # If one enrollment has not been passed, no need to continue,
+                # We are sure that order is not eligible for certification.
+                break
+            passed_enrollment_count += 1
+
+        if passed_enrollment_count != graded_courses_count:
+            return 0
+
+        try:
+            order.create_certificate()
+        except ValidationError:
```

Comment:
what case is this?

Reply:
`ValidationError` is raised if order already have a certificate or if the related product is not certifying.